src/CoreBundle/private/library/classes/dbobjects/TCMSTableConf.class.php

```diff
@@ -152,11 +152,20 @@ public function GetCmsTblListExtension($sExtensionName)
      */
     public function &GetFields(&$oTableRow, $loadDefaults = false, $bDoNotUseAutoObjects = false)
     {
+        // TODO this could also be a parameter here?
```

Comment:
This class derives from TCMSRecord which has a `GetLanguage()` method. I'd suppose that language should be correct.

Reply:
`GetLanguage()` returns the language id - so not directly usable here.

Also `getActiveEditLanguage()` makes it more clear what is meant here.

Reply:
Then we should use the LanguageService to lookup the language object from that ID. If possible, we should use the Tdb's language setting in general and avoid the dependency to the "global" language. This makes it easier to use Tdbs in other contexts (in cases where we'd like to compute things for other languages than the current one).

Reply:
ok

Reply:
GetLanguage() also relies on the/a language being set in the current object.

I'm not sure this is the case in all relevant cases?

Reply:
Checking now if a language exists/is set at all.